### PR TITLE
stopper: wait on double Stop 

### LINKED ISSUE #56728: server: TestDrain failed
[(server).TestDrain failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2447026&tab=buildLog) on [master@5403e542f095bd93f5c805a3ac73f3ebe9c82b4b](https://github.com/cockroachdb/cockroach/commits/5403e542f095bd93f5c805a3ac73f3ebe9c82b4b):

```
=== RUN   TestDrain
    test_log_scope.go:155: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDrain318225734
    test_log_scope.go:64: use -show-logs to present logs inline
    stopper.go:98: leaked stopper, created at:
        goroutine 320243 [running]:
        runtime/debug.Stack(0xc004b2b3b8, 0x428998, 0x30)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/stop.register(0xc0051fabd0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:52 +0x45
        github.com/cockroachdb/cockroach/pkg/util/stop.NewStopper(0x0, 0x0, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0x1ca
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).AddServer(0xc004b12900, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:371 +0xbe
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.NewTestCluster(0x5238620, 0xc0009b2000, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:214 +0x5d8
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.StartTestCluster(0x5238620, 0xc0009b2000, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:130 +0x98
        github.com/cockroachdb/cockroach/pkg/server_test.newTestDrainContext(0xc0009b2000, 0x26)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/drain_test.go:95 +0x85
        github.com/cockroachdb/cockroach/pkg/server_test.doTestDrain(0xc0009b2000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/drain_test.go:40 +0x50
        github.com/cockroachdb/cockroach/pkg/server_test.TestDrain(0xc0009b2000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/drain_test.go:35 +0xb5
        testing.tRunner(0xc0009b2000, 0x4ae0c48)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
--- FAIL: TestDrain (3.53s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestDrain PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestDrain.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

### PR DESCRIPTION
**Pull request by knz**
Fixes #56728 

If there is another `Stop()` already in progress, make the second one
wait on the first one.

### CODE REVIEW

File: pkg/testutils/testcluster/testcluster.go [hunk 1]
```diff
@@ -108,6 +108,11 @@ func (tc *TestCluster) stopServers(ctx context.Context) {
 	for i := range tc.mu.serverStoppers {
 		if tc.mu.serverStoppers[i] != nil {
 			tc.mu.serverStoppers[i].Stop(context.TODO())
+			// We need to check IsStopped() in addition to just waiting for
+			// Stop() above to complete, because if there was an asynchronous
+			// Drain request ongoing, Stop() would take a shortcut and not
+			// actually wait for the async stop to complete.
+			<-tc.mu.serverStoppers[i].IsStopped()
```
Comment (tbg) [inline pkg/testutils/testcluster/testcluster.go:115]: `Stop()` should wait until `IsStopped()` is closed, though. I suggest adding a `<-s.IsStopped()` before the early return in `(*Stopper).Stop`.

Comment (knz) [reply pkg/testutils/testcluster/testcluster.go]: Ok done. Let's see in CI whether there are tests that rely on this 2nd-stop-is-noop behavior.

Comment (tbg) [reply pkg/testutils/testcluster/testcluster.go]: It is a noop :-) Just a bit slower now. I doubt anything will shake out but it's good to check of course.

**Review by tbg — APPROVED** [review]

**Review by knz — COMMENTED** [review]
> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56729)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/testutils/testcluster/testcluster.go, line 115 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56729#-MMGKNu67wPyBawFpzlM-r1-115:-MMH-NJ80oWZFLx6YuoK:b-v96ohv) ([raw file](https://github.com/cockroachdb/cockroach/blob/8a1fe86030f90829c5be730be046371139513654/pkg/testutils/testcluster/testcluster.go#L115)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

It is a noop :-) Just a bit slower now. I doubt anything will shake out but it's good to check of course.
</blockquote></details>

i removed the code in the 2nd commit.

**Comment by knz** [comment]
bors r=tbg

**Pull request merged**